src/plugins/liveobjects/livecounter.ts

```diff
@@ -1,15 +1,112 @@
 import { LiveObject, LiveObjectData } from './liveobject';
+import { LiveObjects } from './liveobjects';
+import { StateCounter, StateCounterOp, StateOperation, StateOperationAction } from './statemessage';
 
 export interface LiveCounterData extends LiveObjectData {
   data: number;
 }
 
 export class LiveCounter extends LiveObject<LiveCounterData> {
+  constructor(
+    liveObjects: LiveObjects,
+    private _created: boolean,
+    initialData?: LiveCounterData | null,
+    objectId?: string,
+  ) {
+    super(liveObjects, initialData, objectId);
+  }
+
   value(): number {
     return this._dataRef.data;
   }
 
+  /**
+   * @internal
+   */
+  isCreated(): boolean {
+    return this._created;
+  }
+
+  /**
+   * @internal
+   */
+  setCreated(created: boolean): void {
+    this._created = created;
+  }
+
+  /**
+   * @internal
+   */
+  applyOperation(op: StateOperation): void {
+    if (op.objectId !== this.getObjectId()) {
+      throw new this._client.ErrorInfo(
+        `Cannot apply state operation with objectId=${op.objectId}, to this LiveCounter with objectId=${this.getObjectId()}`,
+        50000,
+        500,
+      );
+    }
+
+    switch (op.action) {
+      case StateOperationAction.COUNTER_CREATE:
+        this._applyCounterCreate(op.counter);
+        break;
+
+      case StateOperationAction.COUNTER_INC:
+        if (this._client.Utils.isNil(op.counterOp)) {
```

Comment:
i realise this util wasn't added in this PR but the common shorthand javascript terminology for "null or undefined" is 'nullish'

Reply:
I've been using lodash for like 6+ years, so `.isNil` feels right at home for me 😅